### DISCUSSION TITLE
Enable to change Parameters on RootComponent

### DISCUSSION THREAD

**Discussion by Petit-Pas** [dotnet/maui#7918] [Ideas]
I'm fiddling with the MAUI x Blazor thing because I'm really interested in both technologies, and I was looking at sending parameters from MAUI to Blazor component. 

When I do something like: 
```
public MainPage()
{
	InitializeComponent();
    RootComponentControl.Parameters = new Dictionary<string, object> { { "InputString", "input string value" }, { "InputClass", dataContext } };
}
```
in the MAUI component, then 
```
[Parameter]
public string InputString { get; set; }

[Parameter]
public DataContext InputClass{ get; set; }
```
in the Blazor one, everything gets set perfectly. 

However, if I try to later update the Parameter Dictionnary, nothing changed on the Blazor component and the OnParametersSet() function is not being called. 
I saw [here](https://github.com/dotnet/aspnetcore/blob/main/src/Components/WebView/WebView/src/WebViewManager.cs#L216) that the Parameters property is get/init only, and was wondering if it would be a big trouble to enable it to change over time rather than set it into stone. 

Maybe even with a binding from MAUI, detecting any change automatically for instance.

**Comment by mehmetilker** [comment]
Hi @Petit-Pas ,

How did you access from xaml page to BlazorWebView / RootComponentControl control?

**Reply by Petit-Pas** [reply]
Hello @mehmetilker , 

I used the x:Name defined in the XAML page from the C# of the XAML page. actually the same way you see in my MainPage constructor above here, "RootComponentControl" is just the x:Name I did set in the XAML beforehand.

**Comment by mehmetilker** [comment]
Thanks @Petit-Pas it worked. Still trying to figure out going from razor pages to xaml page which contains razor page.

From home.razor
~~~
        Shell.Current.GoToAsync($"{nameof(Category)}?CategoryId=world",
            new Dictionary<string, object>
                {
                    ["Monkey"] = "aaa"
                });
~~~

To Category.xaml

~~~
BlazerViewRoot.Parameters = new Dictionary<string, object> { { "CategoryId", CategoryId } };
~~~

If I set categoryId manually in Category.xaml, I see the value in Category.razor but nothing has been carried from home.razor

Any idea how to to that?

**Reply by Petit-Pas** [reply]
I'm not sure if I understand what you mean, but also it has nothing to do with the original post, maybe open a new topic?

**Reply by mehmetilker** [reply]
No need another topic, since problem still related to carrying parameters to navigated page. 

Setting parameters from constructor did not work but it worked with following way (Although I am still not sure this is the proper way...):


```
    string categoryId;
    public string CategoryId
    {
        get => categoryId;
        set
        {
            categoryId = value;
            OnPropertyChanged();
            BlazerViewRoot.Parameters = new Dictionary<string, object> { { "CategoryId", CategoryId }, { "Bear", bear } };

        }
    }
```
